### PR TITLE
[10.0][ADD] contract_sale

### PR DESCRIPTION
**Pull request by luismontalba**
Contract from Sale
===============
This module allows access to contracts for sale employees without account permissions.
Employees in sale security group "User: Only Own Documents", can only access contracts created by themselves.

Usage
-------
* Log with an user having sale but not account permissions.
* Go to Sales > Sales > Contracts.
* Create a new record or edit another one.

cc @Tecnativa

### CODE REVIEW

File: contract_sale/README.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+==================
+Contract from Sale
+==================
+
+This module allows access to contracts for sale employees without account
+permissions.
+
+Employees in sale security group "User: Only Own Documents", can only access
+ contracts created by themselves.
```
Comment (pedrobaeza) [inline contract_sale/README.rst]: extra space at the beginning

File: contract_sale/security/account_analytic_account_security.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="account_analytic_account_own_salesman" model="ir.rule">
+        <field name="name">See Own Contracts</field>
+        <field name="model_id" ref="analytic.model_account_analytic_account"/>
+        <field name="domain_force">
```
Comment (pedrobaeza) [inline contract_sale/security/account_analytic_account_security.xml]: Put this only in one line for proper visualization in backend UI

File: contract_sale/security/account_analytic_account_security.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (pedrobaeza) [inline contract_sale/security/account_analytic_account_security.xml]: Make this noupdate=1

File: contract_sale/security/account_analytic_account_security.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            ('user_id','=',False),]</field>
+        <field name="groups"
+               eval="[(4,ref('sales_team.group_sale_salesman'))]"/>
+    </record>
+    <record id="account_analytic_account_see_all" model="ir.rule">
+        <field name="name">See All Contracts</field>
+        <field name="model_id" ref="analytic.model_account_analytic_account"/>
+        <field name="domain_force">[(1,'=',1)]</field>
+        <field name="groups"
+               eval="[(4,ref('sales_team.group_sale_salesman_all_leads'))]"/>
```
Comment (pedrobaeza) [inline contract_sale/security/account_analytic_account_security.xml]: Add here also the accounting roles, or they can be cut the permissions if they belong to `group_sale_salesman`

File: contract_sale/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+{
+    'name': 'Contract from Sale',
+    'version': '10.0.1.0.0',
+    'category': 'Sales',
+    'author': 'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/contract',
+    'depends': [
+        'sale',
```
Comment (pedrobaeza) [inline contract_sale/__manifest__.py]: Shouldn't this be enough with `sales_team`?

File: contract_sale/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        'contract',
+    ],
+    'data': [
+        'security/ir.model.access.csv',
+        'security/account_analytic_account_security.xml',
+        'views/account_analytic_account_view.xml',
+    ],
+    'license': 'AGPL-3',
+    'installable': True,
```
Comment (pedrobaeza) [inline contract_sale/__manifest__.py]: Make it auto_install

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
We will need to rebase when main branch turns green

File: contract_sale/__manifest__.py (hunk 5 above)
Comment (luismontalba) [reply contract_sale/__manifest__.py]: Yes, it's enough

**Comment by luismontalba** [comment]
Changes done and branch rebased.

**Comment by luismontalba** [comment]
Travis error seems unrelated:

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Review by chienandalu — COMMENTED** [review]
That travis error is quite similar to the one I've been experiencing in this PR https://github.com/OCA/partner-contact/pull/470. With no solution yet :confused:

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Travis error unrelated.

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

**Pull request merged**